src/back-end/lib/resources/proposal/code-with-us.ts

```diff
@@ -0,0 +1,403 @@
+          });
+        }
+        if (dbResult.value) {
+          return invalid({
+            conflict: ['You already have a proposal for this opportunity.']
+          });
+        }
+
+        const validatedProposalText = proposalValidation.validateProposalText(proposalText);
+        const validatedAdditionalComments = proposalValidation.validateAdditionalComments(additionalComments);
+        const validatedProponent = await validateProponent(connection, proponent);
+        const validatedAttachments = await validateAttachments(connection, attachments);
+
+        if (allValid([validatedProposalText, validatedAdditionalComments, validatedProponent, validatedAttachments])) {
+          return valid({
+            session: request.session,
+            opportunity: validatedCWUOpportunity.value.id,
+            proposalText: validatedProposalText.value,
+            additionalComments: validatedAdditionalComments.value,
+            proponent: validatedProponent.value,
+            attachments: validatedAttachments.value
+          });
+        } else {
+          return invalid({
+            opportunity: getInvalidValue(validatedCWUOpportunity, undefined),
+            proposalText: getInvalidValue(validatedProposalText, undefined),
+            additionalComments: getInvalidValue(validatedAdditionalComments, undefined),
+            proponent: getInvalidValue(validatedProponent, undefined),
+            attachments: getInvalidValue(validatedAttachments, undefined)
+          });
+        }
+      },
+      respond: wrapRespond<ValidatedCreateRequestBody, CreateValidationErrors, JsonResponseBody<CWUProposal>, JsonResponseBody<CreateValidationErrors>, Session>({
+        valid: (async request => {
+          const dbResult = await db.createCWUProposal(connection, omit(request.body, 'session'), request.body.session);
+          if (isInvalid(dbResult)) {
+            return basicResponse(503, request.session, makeJsonResponseBody({ database: [db.ERROR_MESSAGE] }));
+          }
+          return basicResponse(201, request.session, makeJsonResponseBody(dbResult.value));
+        }),
+        invalid: (async request => {
+          return basicResponse(400, request.session, makeJsonResponseBody(request.body));
+        })
+      })
+    };
+  },
+
+  update(connection) {
+    return {
+      async parseRequestBody(request) {
+        const body = request.body.tag === 'json' ? request.body.value : {};
+        const tag = get(body, 'tag');
+        const value: unknown = get(body, 'value');
+        switch (tag) {
+          case 'edit':
+            return adt('edit', {
+              proposalText: getString(value, 'proposalText'),
+              additionalComments: getString(value, 'additionalComments'),
+              proponent: get(value, 'proponent'),
+              attachments: getStringArray(value, 'attachments')
+            });
+          case 'submit':
+            return adt('submit', String(value));
```

Comment:
Change `String(value)` to `getString(body, 'value', '')`. If value is `undefined` or `null`, the current approach will simply store it as `"undefined"` or `"null"`.

Reply:
Same for the cases below.